tests/lib/rules/jsx-curly-spacing.js

```diff
@@ -27,171 +27,1669 @@ var parserOptions = {
   valid: [{
-    code: '<App foo={bar} />;'
+    code: '<App foo={bar}>{bar}</App>;'
+  }, {
+    code: '<App foo={bar}>{ bar }</App>;'
+  }, {
+    code: [
+      '<App foo={',
+      'bar',
+      '}>',
+      '{bar}',
+      '</App>;'
+    ].join('\n')
+  }, {
+    code: '<App foo={{ bar: true, baz: true }}>{{ bar: true, baz: true }}</App>;'
+  }, {
+    code: '<App foo={{ bar: true, baz: true }}>{ { bar: true, baz: true } }</App>;'
   }, {
     code: [
       '<App foo={',
       '{ bar: true, baz: true }',
-      '} />;'
+      '}>',
+      '{{ bar: true, baz: true }}',
+      '</App>;'
     ].join('\n')
   }, {
+    code: '<App>{ foo /* comment */ }</App>'
+  }, {
+    code: '<App>{ /* comment */ foo }</App>'
+  }, {
     code: '<App foo={bar} />;',
-    options: ['never']
+    options: [{attributes: true}]
+  }, {
+    code: [
+      '<App foo={',
+      'bar',
+      '} />;'
+    ].join('\n'),
+    options: [{attributes: true}]
+  }, {
+    code: '<App foo={{ bar: true, baz: true }} />;',
+    options: [{attributes: true}]
   }, {
     code: [
       '<App foo={',
       '{ bar: true, baz: true }',
       '} />;'
     ].join('\n'),
-    options: ['never', {spacing: {objectLiterals: 'never'}}]
+    options: [{attributes: true}]
+  }, {
+    code: '<App foo={bar} />;',
+    options: [{attributes: false}]
+  }, {
+    code: [
+      '<App foo={',
+      'bar',
+      '} />;'
+    ].join('\n'),
+    options: [{attributes: false}]
+  }, {
+    code: '<App foo={{ bar: true, baz: true }} />;',
+    options: [{attributes: false}]
+  }, {
+    code: [
+      '<App foo={',
+      '{ bar: true, baz: true }',
+      '} />;'
+    ].join('\n'),
+    options: [{attributes: false}]
   }, {
     code: '<App foo={ bar } />;',
-    options: ['always']
+    options: [{attributes: false}]
+  }, {
+    code: '<App foo={ { bar: true, baz: true } } />;',
+    options: [{attributes: false}]
+  }, {
+    code: '<App>{bar}</App>;',
+    options: [{children: true}]
+  }, {
+    code: [
+      '<App>{',
+      'bar',
+      '}</App>;'
+    ].join('\n'),
+    options: [{children: true}]
+  }, {
+    code: '<App>{{ bar: true, baz: true }}</App>;',
+    options: [{children: true}]
+  }, {
+    code: [
+      '<App>{',
+      '{ bar: true, baz: true }',
+      '}</App>;'
+    ].join('\n'),
+    options: [{children: true}]
+  }, {
+    code: '<App>{bar}</App>;',
+    options: [{children: false}]
+  }, {
+    code: [
+      '<App>{',
+      'bar',
+      '}</App>;'
+    ].join('\n'),
+    options: [{children: false}]
+  }, {
+    code: '<App>{{ bar: true, baz: true }}</App>;',
+    options: [{children: false}]
+  }, {
+    code: [
+      '<App>{',
+      '{ bar: true, baz: true }',
+      '}</App>;'
+    ].join('\n'),
+    options: [{children: false}]
+  }, {
+    code: '<App>{ bar }</App>;',
+    options: [{children: false}]
+  }, {
+    code: '<App>{ { bar: true, baz: true } }</App>;',
+    options: [{children: false}]
+  }, {
+    code: '<App foo={bar} />;',
```

Comment:
Yes please :-) when existing test cases are utterly untouched, it adds confidence that a change is not breaking.